### ISSUE TITLE
Support for dynamic imports

### ISSUE THREAD

**Issue by gdub22** [benmosher/eslint-plugin-import#700]
https://github.com/tc39/proposal-dynamic-import (stage 3)
```js
import(`./foo/${bar}`).then(module => {
    // ...
})
```
This is currently recognized as invalid syntax: `Parsing error: Unexpected token import`.

Also probably need to add a rule similar to `no-dynamic-require` as well.

**Comment by ljharb** [comment]
The `babel-eslint` parser is required for it to be parsed properly (syntax parsing is not the job of an eslint plugin).

I agree that `no-dynamic-import` is a good rule to have.

**Comment by gdub22** [comment]
Thanks. Using `babel-eslint` resolved my issue.  I incorrectly assumed that using the webpack resolver would apply the babel config it contains.

Leaving open as a request to support a new `no-dynamic-import` rule.

**Comment by benmosher** [comment]
just making a note: would also like to add to the `moduleVisitor` so that static paths imported dynamically (think code splitting) can be linted by `no-unresolved`, `no-absolute-path`, etc.

**Comment by vikr01** [comment]
`no-dynamic-import` is a confusing name since `import()` is called dynamic import.

Also, why make a new rule for this when we can nest it as an option under `no-dynamic-require`?

**Comment by ljharb** [comment]
I think it's pretty easy to ban `import()` with `no-restricted-syntax`, so we probably don't need a rule for that.

**Comment by vikr01** [comment]
@ljharb Do you mean banning with an expression in it?

```javascript
const foopath = './foo';
const foo = await import(foopath);
```

**Comment by ljharb** [comment]
sure, you could do that too.
